modules/nf-core/vclust/cluster/tests/main.nf.test

```diff
@@ -0,0 +1,210 @@
+                ]
+                input[1] = [[:], []]
+                input[2] = false
+                """
+            }
+        }
+    }
+
+    test("sarscov2 - tsv - ani") {
+
+        when {
+            process {
+                """
+                input[0] = VCLUST_ALIGN.out.tsv
+                input[1] = VCLUST_ALIGN.out.ids
+                input[2] = 'ani'
+                input[3] = []
+                input[4] = []
+                input[5] = '0.7'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.log.get(0).get(1)).getText().contains("ani 0.7")},
+                { assert snapshot(
+                    process.out.clusters,
+                    process.out.versions
+                    ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - tsv - tani") {
+
+        when {
+            process {
+                """
+                input[0] = VCLUST_ALIGN.out.tsv
+                input[1] = VCLUST_ALIGN.out.ids
+                input[2] = 'tani'
+                input[3] = 0.89
+                input[4] = []
+                input[5] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.log.get(0).get(1)).getText().contains("tani 0.89")},
+                { assert snapshot(
+                    process.out.clusters,
+                    process.out.versions
+                    ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - tsv - gani") {
+
+        when {
+            process {
+                """
+                input[0] = VCLUST_ALIGN.out.tsv
+                input[1] = VCLUST_ALIGN.out.ids
+                input[2] = 'gani'
+                input[3] = []
+                input[4] = 0.78
+                input[5] = 0.85
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.log.get(0).get(1)).getText().contains("gani 0.78")},
+                { assert snapshot(
+                    process.out.clusters,
+                    process.out.versions
+                    ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - tsv - fail metric") {
+
+        when {
+            process {
+                """
+                input[0] = VCLUST_ALIGN.out.tsv
+                input[1] = VCLUST_ALIGN.out.ids
+                input[2] = []
+                input[3] = []
+                input[4] = 0.78
+                input[5] = 0.85
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.failed },
+                { assert process.errorReport.contains("ERROR: The metric must be specified.")}
```

Comment:
I would still wrap this in a snapshot, that way you can explicitly have the results of every test  in the snapshot (that the report DOES have the string is `true`)